astropy/utils/data.py

```diff
@@ -651,7 +646,7 @@ def get_pkg_data_filename(
 
     Parameters
     ----------
-    data_name : str
+    data_name : str | Path
```

Comment:
This is considered new feature, so probably needs a change log. Same comment applies to all other similar occurrences in this PR (i.e., adding Path support to public API).